src/ripple/protocol/impl/STTx.cpp

```diff
@@ -183,18 +183,19 @@ bool STTx::checkSign(bool allowMultiSign) const
             // at the SigningPubKey.  It it's empty we must be
             // multi-signing.  Otherwise we're single-signing.
             Blob const& signingPubKey = getFieldVL (sfSigningPubKey);
-            sigGood = signingPubKey.empty () ?
+            ret = signingPubKey.empty () ?
                 checkMultiSign () : checkSingleSign ();
         }
         else
         {
-            sigGood = checkSingleSign ();
+            ret = checkSingleSign ();
         }
     }
     catch (std::exception const&)
     {
+        ret = {false, "Internal signature check failure."};
```

Comment:
Nitpick: Since you're explicitly setting `ret` here, I don't think there is any need for the initialization at the top of the function where it's declared.